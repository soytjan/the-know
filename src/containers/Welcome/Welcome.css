@import url("https://fonts.googleapis.com/css?family=Open+Sans:300,400|Poiret+One");
.Welcome {
  height: 100%;
  postion: relative;
  background-image: url("../../assets/concert.jpg");
  background-size: cover; }
