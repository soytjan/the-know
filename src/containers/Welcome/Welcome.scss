@import "../../variables.scss";

.Welcome {
  height: 100%;
  postion: relative;
}

.bg-img {
  background-image: url('../../assets/city-bg.jpg');
  background-size: cover;
}

.half-page {
  height: 50%;
  background-color: $main-bg; 
}