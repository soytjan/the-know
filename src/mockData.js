export const mockEventData = {
  "last_item": null,
  "total_items": "9621",
  "first_item": null,
  "page_number": "1",
  "page_size": "10",
  "page_items": null,
  "search_time": "0.049",
  "page_count": "963",
  "events": {
    "event": [
      {
        "watching_count": null,
        "olson_path": "America/Denver",
        "calendar_count": null,
        "comment_count": null,
        "region_abbr": "CO",
        "postal_code": null,
        "going_count": null,
        "all_day": "0",
        "latitude": "39.7392358",
        "groups": null,
        "url": "http://denver.eventful.com/events/thursday-work-day-july-5-multiple-neighborhoods-/E0-001-111209195-8?utm_source=apis&utm_medium=apim&utm_campaign=apic",
        "id": "E0-001-111209195-8",
        "privacy": "1",
        "city_name": "Denver",
        "link_count": null,
        "longitude": "-104.990251",
        "country_name": "United States",
        "country_abbr": "USA",
        "region_name": "Colorado",
        "start_time": "2018-07-05 09:00:00",
        "tz_id": null,
        "description": " <strong>Sponsored by:</strong> Extreme Community Makeover<br> <strong>Minimum age:</strong> 14<br> Extreme Community Makeover (ECM) is designed to invite volunteers into a partnership with residents of specific Denver neighborhoods to complete home and neighborhood improvement projects. These projects include graffiti removal, landscaping, painting, yard work, or other projects identified by the . . .",
        "modified": "2018-02-19 06:12:10",
        "venue_display": "0",
        "tz_country": null,
        "performers": null,
        "title": "Thursday Work Day, July 5 - Multiple Neighborhoods",
        "venue_address": "Denver, Colorado, United States",
        "geocode_type": "EVDB Geocoder",
        "tz_olson_path": null,
        "recur_string": null,
        "calendars": null,
        "owner": "evdb",
        "going": null,
        "country_abbr2": "US",
        "image": null,
        "created": "2018-02-01 04:56:32",
        "venue_id": "V0-001-000179395-7",
        "tz_city": null,
        "stop_time": null,
        "venue_name": "Denver, Colorado, United States",
        "venue_url": "http://denver.eventful.com/venues/denver-colorado-united-states-/V0-001-000179395-7?utm_source=apis&utm_medium=apim&utm_campaign=apic"
      }
    ]
  }
}

export const mockCleanEventData = [
  {
    title: "Thursday Work Day, July 5 - Multiple Neighborhoods",
    description: " <strong>Sponsored by:</strong> Extreme Community Makeover<br> <strong>Minimum age:</strong> 14<br> Extreme Community Makeover (ECM) is designed to invite volunteers into a partnership with residents of specific Denver neighborhoods to complete home and neighborhood improvement projects. These projects include graffiti removal, landscaping, painting, yard work, or other projects identified by the . . .",
    category: "event",
    venueName: "Denver, Colorado, United States", 
    region: "CO",
    postalCode: null,
    city: "Denver",
    startTime: "2018-07-05 09:00:00",
    venueAddress: "Denver, Colorado, United States",
    image: null,
    url: "http://denver.eventful.com/events/thursday-work-day-july-5-multiple-neighborhoods-/E0-001-111209195-8?utm_source=apis&utm_medium=apim&utm_campaign=apic",
    id: "E0-001-111209195-8"
  }
]