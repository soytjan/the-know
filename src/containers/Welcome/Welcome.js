import React, { Component } from 'react';
// import { connect } from 'react-redux';
// import PropTypes from 'prop-types';
import SearchWelcome from '../SearchWelcome/SearchWelcome';
import './Welcome.css';

// Welcome.propTypes = {

// };

class Welcome extends Component {
  handleReroute = () => {
    return this.props.history.push('/home');
  }

  render() {
    return (
      <section className='Welcome'>
        <section className='bg-img half-page'>
        </section>
        <section className='half-page'>
        </section>
        <SearchWelcome onReroute={this.handleReroute} />
      </section>
    )
  }
}

export default Welcome;