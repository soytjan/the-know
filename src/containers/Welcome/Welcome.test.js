import React from 'react';
import { shallow } from 'enzyme';
import { mockGeolocationData } from '../../mockData';
import { Welcome, mapDispatchToProps } from './Welcome';

describe('Welcome', () => {
  let renderedComponent;
  let mockHistory;
  let mockAddCurrentLocation;  

  beforeEach(() => {
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      status: 200,
      json: () => Promise.resolve(mockGeolocationData)
    }));
    mockHistory = { push: jest.fn() };
    mockAddCurrentLocation = jest.fn();
    renderedComponent = shallow(
      <Welcome 
        history={mockHistory}
        addCurrentLocation={mockAddCurrentLocation}
      />
    );
  });

  it('should match snapshot', () => {
    expect(renderedComponent).toMatchSnapshot();
  });

  it('should call the dispatch function when addCurrentLocation is called from MDTP', () => {
    const mockDispatch = jest.fn();
    const mapped = mapDispatchToProps(mockDispatch);
    mapped.addCurrentLocation();

    expect(mockDispatch).toHaveBeenCalled();
  });

  it('should fetch current location on componentDidMount', () => {
    renderedComponent.instance().componentDidMount();

    expect(window.fetch).toHaveBeenCalled();
  });

  describe('handleReroute', () => {
    it('should call this.props.history.push when handleReroute is called', () => {
      renderedComponent.instance().handleReroute();

      expect(mockHistory.push).toHaveBeenCalled();
    });
  });
});