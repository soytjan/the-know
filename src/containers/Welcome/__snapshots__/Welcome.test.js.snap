// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Welcome
    addCurrentLocation={[Function]}
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="bg-img half-page"
/>,
        <section
          className="half-page"
/>,
        <ErrorBoundary>
          <Connect(SearchWelcome)
                    onReroute={[Function]}
          />
</ErrorBoundary>,
      ],
      "className": "Welcome",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "bg-img half-page",
        },
        "ref": null,
        "rendered": null,
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "half-page",
        },
        "ref": null,
        "rendered": null,
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <Connect(SearchWelcome)
            onReroute={[Function]}
/>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "onReroute": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="bg-img half-page"
/>,
          <section
            className="half-page"
/>,
          <ErrorBoundary>
            <Connect(SearchWelcome)
                        onReroute={[Function]}
            />
</ErrorBoundary>,
        ],
        "className": "Welcome",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "bg-img half-page",
          },
          "ref": null,
          "rendered": null,
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "half-page",
          },
          "ref": null,
          "rendered": null,
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <Connect(SearchWelcome)
              onReroute={[Function]}
/>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onReroute": [Function],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
